modules/tinymce/src/core/main/ts/delete/DeleteUtils.ts

```diff
@@ -67,7 +69,35 @@ const willDeleteLastPositionInElement = (forward: boolean, fromPos: CaretPositio
       }
     }).getOr(true);
 
+const freefallRtl = (root: SugarElement<Node>): Optional<SugarElement<Node>> => {
+  const child = SugarNode.isComment(root) ? Traverse.prevSibling(root) : Traverse.lastChild(root);
+  return child.bind(freefallRtl).orThunk(() => Optional.some(root));
+};
+
+const deleteRangeContents = (editor: Editor, rng: Range, root: SugarElement<HTMLElement>, moveSelection: boolean = true): void => {
+  rng.deleteContents();
+  // Pad the last block node
+  const lastNode = freefallRtl(root).getOr(root);
```

Comment:
Could we put the `getOr` in `freefallRtl` or avoid the `.orThunk`?

Reply:
`getOr` is needed here to actually extract the node
`orThunk` is for handling the haltering case of recursion in `freefallRtl `